build.yaml

```diff
@@ -101,13 +108,12 @@ common_cli:
     # choose to configured.
     # The list of ASCII codes it uses can be found here.
     # https://wiki.archlinux.org/index.php/Color_Bash_Prompt
-    parse_logs() {
+    function parse_logs {
       # ubuntu defaults to mawk, which causes some things to break
```

Comment:
From dash(1):

>    If the program is not a normal executable file (i.e., if
    it does not begin with the "magic number" whose ASCII representation is
    "#!", so execve(2) returns ENOEXEC then) the shell will interpret the
    program in a subshell. The child shell will reinitialize itself in
    this case, so that the effect will be as if a new shell had been
    invoked to handle the ad-hoc shell script, except that the location of
    hashed commands located in the parent shell will be remembered by the
    child.

I bet the script isn't set as executable, so the shell that spawns the script spawns a subshell. This is probably explicitly /bin/sh (via /bin/sh -c <our-script> probably?). So, since /bin/sh is dash on ubuntu, the script is run with that on ubuntu, despite the shebang line. Making the script executable should fix that.